src/aggregator/client/queue.go

```diff
@@ -204,22 +223,61 @@ func (q *queue) Close() error {
 	return nil
 }
 
-func (q *queue) drain() {
+func (q *queue) drain(tC <-chan (time.Time)) {
 	defer q.wg.Done()
+	defer q.conn.Close()
+	lastDrain := time.Now()
+
+	for {
+		select {
+		case qitem := <-q.bufCh:
+			var needFlush bool
+			msg := qitem.Bytes()
+			q.buf = append(q.buf, msg...)
+			qitem.Close()
+
+			if len(q.buf) > q.batchSize {
+				needFlush = true
+			} else {
+				needFlush = time.Since(lastDrain) >= q.writeInterval
```

Comment:
Hm, shouldn't this be the job of the ticker below?  Seems strange to have both potentially do time based flushing (separation of concerns and all).